### ISSUE TITLE
really_destroy! does not hard delete the model relationships

### ISSUE THREAD

**Issue by felipelc** [rubysherpas/paranoia#179]
Take a look at these two classes:

``` ruby
class Lesson < ActiveRecord::Base
    has_many :materials, dependent: :destroy
    acts_as_paranoid
end

class Material < ActiveRecord::Base
    belongs_to :lesson
    acts_as_paranoid
end
```

If I call `lesson.really_destroy!` at a lesson that has many materials, paranoia only hard-delete the lesson. The materials are soft-deleted and stays at DB pointing for a lesson that does not exists anymore.

Can I fix that? I want to recursive "really_destroy" the materials.

Thank you!

**Comment by bluej100** [comment]
I believe this was fixed by https://github.com/radar/paranoia/commit/4f268326c825484b7360dc9145b3ef206153f663 . You'll probably also want https://github.com/radar/paranoia/commit/e7d075f10b80ae77f50d58e39b28bc468e44e4c5 . :-)

If you're seeing this on the tip, could you please provide a test case?

**Comment by felipelc** [comment]
Hi @bluej100 

Use the most updated code at master solves this problem for me. Thank you!

**Comment by radar** [comment]
I've released 2.0.3 today so you don't have to use master anymore
